File: src/optical_element/MATLAB/g_Furion_plane_Mirror.m
classdef  g_Furion_plane_Mirror < g_oe%追迹phase转换成角度
  properties
  
  end
methods
function obj = g_Furion_plane_Mirror( beam_in,ds,di,chi,theta,surface,grating)   
         obj=obj@g_oe( beam_in,ds,di,chi,theta,surface,grating) ;%输入参数给父类    
         obj.reflect(beam_in,ds,di,chi,theta);%反射光束线
end
end
end

File: src/optical_element/g_Furion_plane_Mirror.cpp
#include "g_Furion_plane_Mirror.h"

using namespace Furion_NS;

G_Furion_Plane_Mirror::G_Furion_Plane_Mirror(G_Beam* beam_in, double ds, double di, double chi, double theta, No_Surfe* surface, double r1, double r2, Grating* grating)
    : G_Oe(beam_in, ds, di, chi, theta, surface, grating)
{
    cout << "G_Furion_Plane_Mirror ³õÊ¼»¯" << endl;
}

G_Furion_Plane_Mirror::~G_Furion_Plane_Mirror()
{
    //delete ct;
}

void G_Furion_Plane_Mirror::run(G_Beam* beam_in, double ds, double di, double chi, double theta, No_Surfe* surface, double r1, double r2, Grating* grating)
{
    cout << "G_Furion_Plane_MirrorµÄrun" << endl;
    reflect(beam_in, ds, di, chi, theta);
}